tck/src/main/java/org/eclipse/microprofile/lra/tck/service/spi/LRARecoveryService.java

```diff
@@ -0,0 +1,83 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.eclipse.microprofile.lra.tck.service.spi;
+
+import java.net.URI;
+import java.util.logging.Logger;
+
+/**
+ *     The LRA model guarantees eventual consistency but does not say when
+ *     participants will be brought into a consistent state. When a request
+ *     to close/cancel an LRA is received, the implementation is allowed to
+ *     delay sending out the complete/compensate callbacks (the specification
+ *     only requires that an implementation must eventually send them).
+ *
+ *     This SPI provides a mechanism for implementations to indicate when
+ *     it knows that callbacks have been sent and responses received or when
+ *     callbacks have been issued and all participants have reached an end state.
+ *
+ */
+public interface LRARecoveryService {
+
+    /**
+     * Wait for the delivery of all participant callbacks (Complete or Compensate).
+     * When this method returns the caller can be certain that the callbacks were
+     * sent and responses received (including error responses).
+     *
+     * @param lraId the LRA context
+     * @throws LRACallbackException the implementation was unable to determine whether
+     * or not the callbacks were received by all participants
+     */
+    void waitForCallbacks(URI lraId) throws LRACallbackException;
+
+    /**
+     * Wait for all participants to reach an end state and for all
+     * {@link org.eclipse.microprofile.lra.annotation.AfterLRA} notifications to be delivered.
+     * 
+     * The default implementation iterates {@link LRARecoveryService#waitForEndPhaseReplay(URI)} until
+     * all participants reach a final state.
```

Comment:
"and for all listeners notifications to be successfully delivered."

We either need a separate method to wait for listener notifications or we need make sure the javadoc accurately reflects the behaviour with respect to the listener callbacks.